### PR TITLE
Proposal: forward `Result<T, E>` without need to rewrap

### PR DESCRIPTION
**Pull request by cesarenaldi**
This was something discussed a long time ago and partially available in some 3rd party implementations of the Result monad.

In this PR you can see the change I am suggesting to the `Result` definition, which are backwards compatible as well as some places to showcase the impact.

If the proposal gets approved I will amends the rest of the codebase to avoid rewrapping in `success(...)`, `failure(...)`.

### CODE REVIEW

**Review by reecejohnson — APPROVED** [review]
this is lovely!

File: packages/shared-kernel/src/Result.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { Narrow } from './ts-helpers/types';
  * @sealed
  * @privateRemarks DO NOT EXPORT, see type export later on
  */
-class Success<T, E> {
+class Success<T, E extends IEquatableError> {
```
Comment (desfero) [inline packages/shared-kernel/src/Result.ts:13]: If we can afford the breaking change and removal of the second generic would simplify to `Success<T>` and `Failure<E>`. In this case you dont' even need `forward` as from the type perspective it should match the value correctly.

Comment (cesarenaldi) [reply packages/shared-kernel/src/Result.ts:13]: I think it breaks the relationship with the underlying error/data?
image

Comment (cesarenaldi) [reply packages/shared-kernel/src/Result.ts:13]: even worse I get this:
image

Comment (cesarenaldi) [reply packages/shared-kernel/src/Result.ts:13]: Ignore me.

**Comment by cesarenaldi** [comment]
Closing this in favour of: https://github.com/lens-protocol/lens-sdk/pull/575

**Pull request closed without merging**